Add Saving Satoshi case study (#1131)
BitcoinDesign/Guide: guide/case-studies/saving-satoshi.md

The universe really came to life at the first Saving Satoshi workshop where swag was created to represent different elements of the game like logos for the mining groups featured in chapter 3 and holographic Holocat stickers.

{% include picture.html
   image = "assets/images/guide/case-studies/saving-satoshi/swag.jpg"
   retina = "assets/images/guide/case-studies/saving-satoshi/swag@2x.jpg"
   mobile = "assets/images/guide/case-studies/saving-satoshi/swag-mobile.jpg"
   mobileRetina = "assets/images/guide/case-studies/saving-satoshi/swag-mobile@2x.jpg"
   alt-text = "Collection of Saving Satoshi merchandise including holographic cat stickers, 'I saved Satoshi' stickers, 'Thomas Vanderpoole is not Satoshi' labels, and Creed CD-ROM stickers."
   caption = "Swag created for the TABConf 2024 workshop."
   width = 800
   height = 503
   layout = "full-width"
%}

While there were a number of things that didn’t make it into the final cut of the story these ideas have been saved for later releases. Look forward to a future update with a list of bitcoin themed cocktail drinks and much more!

> **Incorporating bitcoin culture and history**

Reviewer: I'm not sure why but there is no line break after this in the preview

Author: Good catch. I fixed those issues (with extra line breaks). Main problem is that this is bold text, when it should be a header. We have several other instances like this. I'll take a look tomorrow and update if necessary.

Reviewer: Ah, got it! I have no opinion on the formatting for this (bold or header) so feel free to change as you see fit!

Author: Headers are the better semantic HTML choice, so I switched to those. Should all look as expected now.

Mentions of bitcoin culture and history can be found all over the game. The very first exercise in the game is to find the secret message in the genesis block (“The Times 03/Jan/2009 Chancellor on brink of second bailout for banks”). At several points in the game “don’t trust, verify” is taken quite literally like chapter 8 when users compute block subsidies and check that the longest chain adheres to the rules. When bitcoin Script is introduced in chapter nine, Holocat delivers a message about the ```OP_CHECKMULTISIG``` bug in Satoshi’s code that pops an extra item off the stack. In chapter ten players meet a character named Laszlo, a nod to the famous Laszlo Hanyecz who bought two pizzas for 10,000 bitcoin in 2020. Even the spirit of the game, one where the protagonist can be anybody but still save bitcoin, echoes the popular saying, “We are all Satoshi”.

## 3. Design

Powered by an immersive game UX, Saving Satoshi uses rich visuals and seamless interactions to journey players through time and space to learn about bitcoin coding. However, that is much easier said than done and the blending of story and education resulted in some unique design challenges.

> An early ambition proposed by Christoph was to illustrate the story using generative AI. Compared to what is available today, the AI image generation tools of late 2022 were rudimentary. There was no shortage of issues including things like six fingered hands, inconsistent characters, limited resolutions, and little to no control over an image’s composition. Despite these inconveniences we stayed committed to this direction. There was just an expectation that we would have to iterate as the technology progressed. Fast forward to the end of chapter ten and that’s exactly what happened. While you may still see an extra finger or two, it’s clear that the quality of the art has made leaps and bounds. Christoph’s foresight in choosing this direction cannot be understated. The strategy worked out well and allowed the game’s visual storytelling to make huge strides.

Reviewer: ```suggestion
An early ambition proposed by Christoph was to illustrate the story using generative AI. Compared to what is available today, the AI image generation tools of late 2022 were rudimentary. There was no shortage of issues including things like six fingered hands, inconsistent characters, limited resolutions, and little to no control over an image’s composition. Despite these inconveniences we stayed committed to this direction. There was just an expectation that we would have to iterate as the technology progressed. Fast forward to the end of chapter ten and that’s exactly what happened. While you may still see an extra finger or two, it’s clear that the quality of the art has made leaps and bounds. Christoph’s foresight in choosing this direction and betting on the future cannot be understated. The strategy worked out well and allowed the game’s visual storytelling to make huge strides.
```

## What’s next

Chapter 10 may be a major milestone for Saving Satoshi, but the project is far from over! At the time of writing, December 2024, there are still many things the team has planned for future releases. From design, to learning content, to community, these ideas are not confined to just one aspect of the project and instead span across many facets of the game.

**Difficulty levels**
One of the larger items on the to-do list is support for difficulty levels. In an ideal world, Saving Satoshi would be able to accommodate a variety of learners but currently, a solid understanding of how to code is a major requirement. It would be nice to one day support people that are just getting started and are new to coding. Conversely, it would also be great to make advanced exercises for those looking for more of a challenge. This has been implemented to some degree by sometimes moving hints to a separate help page, but there is so much more that can be done. At the moment, chapter six supports two difficulty modes (normal and hard). This was initially implemented as a proof-of-concept and the team looks forward to extending this functionality to other chapters.

> **Movie trailer**

Reviewer: One more - Same issue here - I'm not sure why but there is no line break after this in the preview

It’s a labor of love.

We encourage anyone interested in learning more to play the game and join the community building Saving Satoshi. To get in touch, follow [Saving Satoshi on GitHub](https://github.com/saving-satoshi/saving-satoshi) and join the [#saving-satoshi channel](https://discord.gg/B9WURkwF) on the Bitcoin Design Discord.

Now be quick! [Satoshi needs you](https://savingsatoshi.com).

### Acknowledgements

Saving Satoshi would not be possible without the many people that have come together over the last few years to build it. The core team is made up of Adam Jonas, Christoph Ono (design lead), Satsie, BenAllenG (engineering lead), Matthew Zipkin (lesson content lead), Tuedon Tuoyo, Art Assoiants (product manager), and 4 mysterious cats that Spiral let us borrow to write the story.

> Throughout the journey we have also had a number of contributors play key roles at various points in the game. This includes, but is not limited to, ecurrencyhodler, JT Woodhouse, JSMS, Kevin Karsopawiro, daviroo, Vinayak Sharma, Shashwat Vangani, Aurèle Oulès, and sabakhilji.

Reviewer: ```suggestion
Throughout the journey we have also had a number of contributors play key roles at various points in the game. This includes, but is not limited to, ecurrencyhodler, JT Woodhouse, JSMS, Kevin Karsopawiro, daviroo, Vinayak Sharma, Shashwat Vangani, Aurèle Oulès, and sabakhilji. We'd also like to thank the countless testers that tried out the game and provided invaluable feedback on their experience.
```

   caption = "Some of the unique interface components built for the learning experiences."
   width = 1200
   height = 704
   layout = "full-width"
%}

Third, it was important for the team to support smartphones. Many people around the world have one computer - their phone. A separate navigation and layout logic was designed to accommodate both mobile users and give them full access to the whole experience.

To peek into the design process, you can find the main design file [here](https://www.figma.com/design/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=0-1&t=hKCIATRVZnYLooMh-1). The process for image generation is documented [here](https://www.figma.com/design/0Fg2IiEbLI5o8juawKqkJM/Saving-Satoshi-images?node-id=407-2995&t=3iZ1Yh3n3bOCcBaM-1). Plus, Christoph created [various recordings](https://www.youtube.com/@germanysbestkeptsecret/search?query=saving%20satoshi) of the design process.

## 4. Development

A big challenge in development has been thinking about incrementing difficulty and faithfulness to the realistic logic of bitcoin code.

> We found creating new and unique components important to keeping the experience interesting while trying to keep development time as low as possible.

Reviewer: ```suggestion
We found that experimenting with new and unique components was key to finding the best ways to teach content and let users interact with the material. One thing to keep in mind was balancing this with the desire to keep development time as low as possible.
```

**Movie trailer**

On the design and marketing front, a big dream is to do a movie trailer (here’s an [early exploration](https://www.youtube.com/watch?v=EW6tmJMNc5g) from 2023) and poster. When the project first began, the AI image and video generation tools were nowhere near where they had to be to support something like this. A lot has changed in the last two years! Even just over the last few months, the design team has made some very impressive short clips to demonstrate what the technology today is capable of. It’s only going to get better and as it does, the possibility of a movie trailer becomes more likely.

**Community involvement**

> In terms of community, the team would like to put more effort towards tapping into the larger bitcoin and software engineering ecosystems to share the game and partner with neighboring groups. More educator tools like a guide for individuals that want to run Saving Satoshi workshops or a comprehensive list of topics that the game covers would help with this initiative.

Reviewer: ```suggestion
In terms of community, the team would like to put more effort towards tapping into the larger bitcoin and software engineering ecosystems to share the game and partner with neighboring groups. If this sounds like you, hop into our [Discord channel](https://discord.gg/B9WURkwF) and get in touch with us! More educator tools like a guide for individuals that want to run Saving Satoshi workshops or a comprehensive list of topics that the game covers would help with this initiative.
```

## Conclusion

By design, Saving Satoshi is reflective of the best parts of bitcoin. It has endless rabbit holes. While a cultural transmission in itself, it also brings forth a rich and vibrant world of its own. It’s permissionless and makes players think in ways that they may not be used to. Like bitcoin, it’s 100% open source and built by a passionate group of contributors under the highest standards. It’s by the people, for the people.

It’s a labor of love.

We encourage anyone interested in learning more to play the game and join the community building Saving Satoshi. To get in touch, follow [Saving Satoshi on GitHub](https://github.com/saving-satoshi/saving-satoshi) and join the [#saving-satoshi channel](https://discord.gg/B9WURkwF) on the Bitcoin Design Discord.

Now be quick! [Satoshi needs you](https://savingsatoshi.com).

### Acknowledgements

> Saving Satoshi would not be possible without the many people that have come together over the last few years to build it. The core team is made up of Adam Jonas, Christoph Ono (design lead), Satsie, BenAllenG (engineering lead), Matthew Zipkin (lesson content lead), Tuedon Tuoyo, Art Assoiants (product manager), and 4 mysterious cats that Spiral let us borrow to write the story.

Reviewer: ```suggestion
Saving Satoshi would not be possible without the many people that have come together over the last few years to build it. The core team is made up of [Adam Jonas](https://adamjonas.com/), [Christoph Ono](https://www.germanysbestkeptsecret.com/) (design lead), [Satsie](https://satsie.dev), [BenAllenG](https://github.com/benalleng) (engineering lead), [Matthew Zipkin](https://github.com/pinheadmz) (lesson content lead), [Tuedon Tuoyo](https://x.com/TuedonTuoyo), [Art Assoiants](https://www.artassoiants.com/) (product manager), and 4 mysterious cats that [Spiral](https://spiral.xyz) let us borrow to write the story.
```

Author: This is updated in my latest commit (did not see your comment here until it was too late).

> Throughout the journey we have also had a number of contributors play key roles at various points in the game. This includes, but is not limited to, ecurrencyhodler, JT Woodhouse, JSMS, Kevin Karsopawiro, daviroo, Vinayak Sharma, Shashwat Vangani, Aurèle Oulès, and sabakhilji. We'd also like to thank the countless testers that tried out the game and provided invaluable feedback on their experience.

Reviewer: ```suggestion
Throughout the journey we have also had a number of contributors play key roles at various points in the game. This includes, but is not limited to, [ecurrencyhodler](https://x.com/ecurrencyhodler), [JT Woodhouse](https://www.jtwoodhouse.com/), [JSMS](https://x.com/_Jsms_), [Kevin Karsopawiro](https://github.com/afterburn), [daviroo](https://github.com/daviroo), [nully0x](https://github.com/nully0x), [Vinayak Sharma](https://github.com/vinayaksh42), [Shashwat Vangani](https://github.com/shaavan), [Aurèle Oulès](https://github.com/aureleoules), and [sabakhilji](https://github.com/sabakhilji). We'd also like to thank the countless testers that tried out the game and provided invaluable feedback on their experience.
```
